src/roles/spook.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+const MetaRole = require('roles_meta')
+const CONTROLLER_MESSAGE = '* Self Managed Code * quorum.tedivm.com * #quorum in Slack *'
+
+class Spook extends MetaRole {
+  getBuild (room, options) {
+    return [MOVE]
+  }
+
+  manageCreep (creep) {
+    // Delete current target if it is currently visible.
+    if (creep.memory.starget && Game.rooms[creep.memory.starget]) {
+      delete creep.memory.starget
+    }
+
+    // Harass current room by signing controller and stomping contruction sites.
+    if (!creep.room.controller || !creep.room.controller.my) {
+      if (!creep.memory.stomproom || creep.memory.stomproom !== creep.room.name) {
+        creep.memory.stomproom = creep.room.name
+        creep.memory.stomptime = Game.time
+      }
+
+      if (Game.time - creep.memory.stomptime < 70) {
+        // Find and destroy any hostile construction sites
+        if (this.stompConstruction(creep)) {
+          return
+        }
+        // Tag the controller
+        if (this.signController(creep)) {
+          return
+        }
+      }
+    }
+
+    // Move on to the next room
+    this.migrateRooms(creep)
+  }
+
+  stompConstruction (creep) {
+    const construction = creep.pos.findClosestByRange(FIND_HOSTILE_CONSTRUCTION_SITES)
```

Comment:
if we don't filter out construction sites in the same position as the creep then the creep can get stuck on sites that are instantly replaced

Reply:
I thought you couldn't place a construction site where a hostile creep was standing?

Reply:
I've had stomping creeps get stuck that way before at least.